docs/csharp/language-reference/statements/fixed.md

```diff
@@ -0,0 +1,64 @@
+---
+title: "fixed statement - C# reference"
+description: "Use the C# fixed statement to pin a moveable variable for the duration of the statement."
+ms.date: 09/09/2022
+f1_keywords: 
+  - "fixed_CSharpKeyword"
+  - "fixed"
+helpviewer_keywords: 
+  - "fixed statement [C#]"
+  - "fixed keyword [C#]"
+---
+# fixed statement (C# reference)
+
+The `fixed` statement prevents the [garbage collector](../../../standard/garbage-collection/index.md) from relocating a moveable variable and declares a pointer to that variable:
+
+:::code language="csharp" source="snippets/fixed/Program.cs" id="PinnedArray":::
+
+The address of a fixed, or pinned, variable doesn't change for the duration of the statement. You can use the declared pointer only inside the corresponding `fixed` statement. The declared pointer is readonly and cannot be modified.
+
+> [!NOTE]
+> You can use the `fixed` statement only in an [unsafe](../keywords/unsafe.md) context. The code that contains unsafe blocks must be compiled with the [**AllowUnsafeBlocks**](../compiler-options/language.md#allowunsafeblocks) compiler option.
```

Comment:
@BillWagner question not for this PR, but for possible upcoming work: what do you think about the article about `unsafe`:
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/unsafe

Now, it's in TOC under "Keywords -> Modifiers" as `unsafe` may be used as a type/method modifier. Do you want the dedicated article about `unsafe` statement under "Statements" in TOC? If yes, then that article would be very short and I guess the current article under modifiers still should stay. What do you think?

Reply:
Hi @pkulikov 

In time, I'd like to have one area of the **Language Reference** section devoted to *unsafe code* including `fixed`, `unsafe`, pointer types, and so on.  My reasoning is that the recent language enhancements around `ref` (including the `ref` fields and `scoped` additions) mean that it's much less likely for developers to need any unsafe constructs. I'd like to put those constructs together in one section.

Reply:
@BillWagner ok, clear, thanks. Then, is it already fine in meantime to "merge" the `unsafe` keyword article into the [Unsafe code article](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/unsafe-code) as one more section in it?

Reply:
@pkulikov 

> is it already fine in meantime to "merge" the unsafe keyword article into the [Unsafe code article](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/unsafe-code) as one more section in it?

Yes. Thanks for asking.